core/templates/pages/topics-and-skills-dashboard-page/modals/delete-skill-modal.component.ts

```diff
@@ -43,28 +44,43 @@ export class DeleteSkillModalComponent extends ConfirmOrCancelModal {
     super(ngbActiveModal);
   }
 
+  fetchTopicIdToDiagnosticTestSkillIds(
+      topicAssignments: AssignedSkill[]): void {
+    let allTopicIds = [];
+    for (let topic of topicAssignments) {
+      allTopicIds.push(topic.topicId);
+    }
+    let topicIdsNotEligibleForUnassignment: string[] = [];
+    this.topicsAndSkillsDashboardBackendApiService
+      .fetchTopicIdToDiagnosticTestSkillIdsAsync(allTopicIds).then(
+        (dict) => {
+          for (let topicId in dict.topicIdToDiagnosticTestSkillIds) {
+            let diagnosticTestSkillIds = (
+              dict.topicIdToDiagnosticTestSkillIds[topicId]);
+            if (diagnosticTestSkillIds.length === 1 &&
+                diagnosticTestSkillIds.indexOf(this.skillId) !== -1) {
+              this.allowDeletionOfSkill = false;
+              topicIdsNotEligibleForUnassignment.push(topicId);
+            }
+          }
+          topicAssignments.filter((topic) => {
+            if (topicIdsNotEligibleForUnassignment.indexOf(
+              topic.topicId) === -1) {
+              return true;
+            }
+            this.topicNamesNotEligibleForUnassignment.push(topic.topicName);
+            return false;
+          });
+          this.topicsAssignments = topicAssignments;
+          this.topicsAssignmentsAreFetched = true;
+        });
+  }
+
   fetchTopicAssignmentsForSkill(): void {
     this.topicsAndSkillsDashboardBackendApiService
       .fetchTopicAssignmentsForSkillAsync(
         this.skillId).then((response: AssignedSkill[]) => {
```

Comment:
Break before the first ')' and deindent the following continuation line by 2. Then line 83 (which becomes line 84) can be indented by 2 from line 82 (which becomes line 83), as expected due to the '{'.

Reply:
updated